include/zephyr/spinlock.h

```diff
@@ -199,16 +227,34 @@ static ALWAYS_INLINE int k_spin_trylock(struct k_spinlock *l, k_spinlock_key_t *
 
 	z_spinlock_validate_pre(l);
 #ifdef CONFIG_SMP
+#ifdef CONFIG_TICKET_SPINLOCKS
+	atomic_val_t ticket_val = atomic_get(&l->owner);
+
+	/*
+	 * For real-life systems the spinlock values will not be able to wrap
+	 * around to erroneously match the ticket_val, so if the CAS succeeds
```

Comment:
Is that... really true?  What about a malicious attacker?  Four billion tries is actually quite reasonable to do...

I guess the scenario I'm thinking of would be an API exposed to a userspace process that allowed high-bandwidth parallel locking, and tricked a subsystem into wrapping its lock, allowing a race that then corrupted internal data, etc...

(But FWIW: this version looks otherwise much better to me than the original, which I still think had a race with the unconditional write)